### ISSUE TITLE
Bug Report: [Character's offsets change when hitting notes if their scale is different from 1]

### ISSUE THREAD

**Issue by Venesio** [FunkinCrew/Funkin#2543]
### If you are playing the game in a browser, what site are you playing it from?

- [ ] [Newgrounds](https://www.newgrounds.com/portal/view/770371)
- [X] [Itch.io](https://ninja-muffin24.itch.io/funkin)? Specify below
- 
  - [X] Windows
- 
  - [ ] Mac
- 
  - [ ] Linux

### If you are playing the game in a browser, what browser are you using?

- [ ] Google Chrome (or chomium based like Brave, vivaldi, MS Edge)
- [ ] Firefox
- [ ] Safari

## What version of the game are you using? Look in the bottom left corner of the main menu. (ex: 0.2.7, 0.2.1, shit like that)

v0.3.2

## Have you identified any steps to reproduce the bug? If so, please describe them below in as much detail as possible. Use images if possible.

![image](https://uploads.linear.app/b9f0050d-7020-42ad-9d4b-7551ff30d00f/70c75900-e2d6-428a-8b64-86d3650750d7/f9ac420e-6dcd-4ace-b384-5bab65c07de2?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2I5ZjAwNTBkLTcwMjAtNDJhZC05ZDRiLTc1NTFmZjMwZDAwZi83MGM3NTkwMC1lMmQ2LTQyOGEtOGI2NC04NmQzNjUwNzUwZDcvZjlhYzQyMGUtNmRjZC00YWNlLWIzODQtNWJhYjY1YzA3ZGUyIiwiaWF0IjoxNzE3Nzc1Njc1LCJleHAiOjMzMjg4MzM1Njc1fQ.GqrwhhPPBqwWH3dkz10y9EfcmqQN5HF9nY3uGcJLxrA)

[https://github.com/FunkinCrew/Funkin/assets/97074204/1ab2c63a-461c-4a48-8414-bf21c3e76081](https://github.com/FunkinCrew/Funkin/assets/97074204/1ab2c63a-461c-4a48-8414-bf21c3e76081)

## Please describe your issue. Provide extensive detail and images if possible.

If the character's scale is different from 1 in their JSON file, their offsets change with each animation and they start flying to another direction.

## If you're game is FROZEN and you're playing a web version, press F12 to open up browser dev window, and go to console, and copy-paste whatever red error you're getting

**Comment by objShaggy** [comment]
Such things are always bound to happen due to offsets not being calculated with the scale and I don't think they're ever gonna fix this.

**Comment by VocalFan** [comment]
https://github.com/gusborg88/fnf-porter

We're encountering this issue a LOT when we import the scale value from Psych Engine characters to Base Game that use the scale value.

**Comment by SarahDrewington** [comment]
Really hope this issue gets fixed.

**Comment by Mlaofmd** [comment]
You can fix that

go to ... `yourmod/scripts/characters/` and create your character script `yourcharactername.hxc`

and copy this, but change names!
```haxe
import funkin.play.character.SparrowCharacter;

class YourCharNameCharacter extends SparrowCharacter {
    function new() {
        super('yourcharacter');
    }
    
	// script by Nebula_Zorua
	// retry fix by TormentedProgram
	// please leave this credit in
	
    override function set_animOffsets(value:Array<Float>) {
        if(animOffsets == null)
            animOffsets = [0, 0];
        
        if(animOffsets[0] == value[0] && animOffsets[1] == value[1])
            return value;

        var scaleX = scale.x / (isPixel ? 6 : 1);
        var scaleY = scale.y / (isPixel ? 6 : 1);

        var scaledNewX = value[0] * scaleX;
        var scaledNewY = value[1] * scaleY;
        var scaledOldX = animOffsets[0] * scaleX;
        var scaledOldY = animOffsets[1] * scaleY;
        var xDiff = scaledOldX - scaledNewX;
        var yDiff = scaledOldY - scaledNewY;

        x = x + xDiff;
        y = y + yDiff;

        animOffsets[0] = value[0];
        animOffsets[1] = value[1];

        return value;
    }

    override function onCountdownStart(event) {
        super.onCountdownStart(event);
        setScale(_data.scale); 	//scale breaks on retry this fixes that
    }
}
```

**Comment by EliteMasterEric** [comment]
Pretty sure this is fixed in v0.5
